internal/http/parser/http_response.go

```diff
@@ -0,0 +1,225 @@
+package http_parser
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+)
+
+type Header struct {
+	Key   string
+	Value string
+}
+
+type StatusLine struct {
+	Version    string
+	StatusCode int
+	Reason     string
+}
+
+type HTTPResponse struct {
+	StatusLine StatusLine
+
+	Headers []Header
+
+	Body []byte
+}
+
+func Parse(data []byte) (httpResponse HTTPResponse, readBytesCount int, err error) {
+	reader := bytes.NewReader(data)
+
+	response, err := doParseResponse(reader)
+	if err != nil {
+		return HTTPResponse{}, 0, err
+	}
+
+	return response, len(data) - reader.Len(), nil
+}
+
+func doParseResponse(reader *bytes.Reader) (HTTPResponse, error) {
+	var allHeadersFound = false
+	var sectionsFound int
+	var R HTTPResponse
+	var SL StatusLine
+
+	offsetBeforeCurrentSection := GetReaderOffset(reader)
+	versionB, err := ReadUntilAnyDelimiter(reader, [][]byte{SPACE, TAB})
+	if err == io.EOF {
+		return HTTPResponse{}, IncompleteInputError{
+			Reader:  reader,
+			Message: fmt.Sprintf("Expected: HTTP-version, Received: %q", versionB),
```

Comment:
How will the user know what "HTTP-version" is? Let's find a more friendlier error message. 

The logic here could be far simpler - read exactly 8 bytes, and check it it matches `HTTP/1.1`. Not sure why we have to read will a SPACE/TAB separator